### PR TITLE
Lsra doc

### PR DESCRIPTION
**Pull request by CarolEidt**
Update LinearScan section of ryujit-overview, and add lsra-detail

### CODE REVIEW

File: Documentation/design-docs/lsra-detail.md [hunk 1]
```diff
@@ -0,0 +1,973 @@
...
+            above, a `RefTypeBB` `RefPosition` is created at each
+            new `BasicBlock`. This is the signal to the register
+            allocator to re-establish variable locations at the
+            boundary.
+
+        -   Iterating over the nodes within the block in execution
+            order:
+
+            -   Two locations (`LsraLocation`, which is an `unsigned
+                int`) are assignedto each node.
```
Comment (mikedn) [inline Documentation/design-docs/lsra-detail.md]: missing space in assigned to

File: Documentation/botr/ryujit-overview.md [hunk 2]
```diff
@@ -379,27 +379,50 @@ The “@ 15” is the location number of the node.  The “0=1” indicates that
...
+* All nodes for which a register is optional are identified (`gtLsraInfo.regOptional` is `true`)
+  * This is used for x86 and x64 on operands that can be directly consumed from memory if no register is allocated.
+* All unused values (nodes that produce a result that is not consumed) are identified (gtLIRFlags has the LIR::Flags::UnusedValue bit set)
+  * Since tree temps (the values produced by nodes and consumed by their parent) are expected to be single-def, single-use (SDSU), normally the live range can be determined to end at the use. If there is no use, the register allocator doesn't know where the live range ends.
+* Code can be generated without any context from the parent (consumer) of each node.
+
+After `Lowering` has completed, liveness analysis is performed:
+* It identifies which `lclVar`s should have their liveness computed.
+  * The reason this is done after `Lowering` is that it can introduce new `lclVar`s.
+* It the does liveness analysis on those `lclVar`s, updating the `bbLiveIn` and `bbLiveOut` sets for each `BasicBlock`.
```
Comment (sandreenko) [inline Documentation/botr/ryujit-overview.md]: It ~the~ does

File: Documentation/design-docs/lsra-detail.md [hunk 3]
```diff
@@ -0,0 +1,973 @@
...
+            registers, volatile/caller-save, write barriers, dedicated
+            registers)
+
+-   Iterate over RefPositions in forward order, performing linear scan
+    allocation. The algorithm below is a modified version of that
+    specified in [[3]](#[3]), and doesn't include optimizations such as early
+    exit of the loops when the remaining intervals will be out of range.
+    It is a high-level abstraction of the algorithm as implemented:
+```
+LinearScanAllocation(List\<RefPosition\> refPositions)
```
Comment (fiigii) [inline Documentation/design-docs/lsra-detail.md]: Code block does not need the escape backslash.

**Comment by CarolEidt** [comment]
I incorporated the feedback, and did some further cleanup. I expect this to be an evolving document.

**Pull request merged**